eureka/src/main/java/com/linecorp/armeria/server/eureka/EurekaUpdatingListener.java

```diff
@@ -328,26 +330,36 @@ public void run() {
             client.sendHeartBeat(appName, instanceId, instanceInfo, null)
                   .aggregate()
                   .handle((res, cause) -> {
-                      try (HttpData content = res.content()) {
-                          if (closed) {
-                              return null;
-                          }
+                      if (closed) {
+                          return null;
+                      }
 
-                          // The information of this instance is removed from the registry when the heart beats
-                          // fail consecutive three times, so we don't retry.
+                      if (cause != null) {
+                          logger.warn("Failed to send a heart beat to Eureka: {}", client.uri(), cause);
+                      } else if (res.headers().status() != HttpStatus.OK) {
```

Comment:
nit: How about introducing `res.headers().status()` as a local variable and cleaning up `if else` condition?
```java
if (cause != null) {

} else {
   HttpStatus status = res.status();
   if (status == HttpStatus.OK) {
        // schedule heart beat 
   } else if (status == HttpStatus.NOT_FOUND) {
        // re-register
   } else {
       // leave warn message
   }
}